my-app/src/pages/Delivery/ClusterMap.tsx

```
@@ -493,15 +493,53 @@ const ClusterMap: React.FC<ClusterMapProps> = ({ visibleRows, clusters, clientOv
+            if (clusterSelect) initialClusterId = clusterSelect.value;
+            if (driverSelect) initialDriver = driverSelect.value;
+            if (timeSelect) initialTime = timeSelect.value;
             viewMode.style.display = 'none';
             editMode.style.display = 'block';
+            // Ensure popup stays in view after switching to edit mode
+            setTimeout(() => {
+              // Find the marker for this popup
+              if (markerGroupRef.current && mapRef.current) {
+                markerGroupRef.current.eachLayer(function(layer) {
+                  // Only proceed if this is a marker
+                  if ((layer as L.Marker).getPopup && (layer as L.Marker).getPopup() && (layer as L.Marker).getPopup()!.getContent() === popupContainer) {
```

Biome (1.9.4)

[error] Change to an optional chain.

Unsafe fix: Change to an optional chain.


(lint/complexity/useOptionalChain)